utils/build-windows.bat

```diff
@@ -0,0 +1,287 @@
+:: - CMAKE_BUILD_TYPE: Kind of build: Release, RelWithDebInfo, Debug.
+:: - PYTHON_HOME: The Python installation directory.
+
+:: REQUIRED PERMISSIONS
+:: Practically, it is easier to be in the Adminstrators group to run the
+:: script, but it should be possible to execute as a normal user.
+:: The user will need permission to write files into the Windows SDK and the
+:: VisualC++ folder.
+
+@echo off
+
+setlocal enableextensions enabledelayedexpansion
+
+set "exitOnError=|| (exit /b)"
+set current_directory=%~dp0
+set current_directory=%current_directory:~0,-1%
+set source_root=%current_directory%\..\..
+
+cd %source_root%
+set source_root=%CD%
+set build_root=%source_root%\build
+set install_directory=%build_root%\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr
+
+mkdir %build_root%
+
+call :clone_repositories %exitOnError%
+call :download_icu %exitOnError%
+:: TODO: Disabled until we need LLBuild/SwiftPM in this build script.
+:: call :download_sqlite3
+
+call :build_llvm %exitOnError%
+path %PATH%;%install_directory%\bin
+
+call :build_cmark %exitOnError%
+
+call :prepare_platform_modules %exitOnError%
+call :build_swift %exitOnError%
+
+call :build_lldb %exitOnError%
+
+path %source_root%\icu-63.1\bin64;%install_directory%\bin;%build_root%\swift\libdispatch-prefix\bin;%PATH%;%ProgramFiles%\Git\usr\bin
+call :test_swift %exitOnError%
+
+goto :end
+endlocal
+
+:clone_repositories
+:: Clones the repositories used by the Windows build.
+:: It supposes that the swift repository is already cloned by CI.
+:: It supposes the current_directory is the parent of the swift repository.
+setlocal enableextensions enabledelayedexpansion
+
+git clone --depth 1 --single-branch https://github.com/apple/swift-cmark cmark %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-clang clang %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-llvm llvm %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-compiler-rt compiler-rt %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-corelibs-libdispatch %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-corelibs-foundation %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-corelibs-xctest %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-lldb lldb %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-llbuild llbuild %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-package-manager %exitOnError%
+git clone --depth 1 --single-branch https://github.com/curl/curl.git %exitOnError%
+git clone --depth 1 --single-branch https://gitlab.gnome.org/GNOME/libxml2.git %exitOnError%
+
+goto :eof
+endlocal
+
+
+:download_icu
+:: Downloads SQLite3, which will be used as a dependency for llbuild and
```

Comment:
duplicated description from `download_sqlite3` bellow

Reply:
Thanks. I fixed this comment and one in the clone repository that was also incorrect.